### DISCUSSION TITLE
Does Langfuse tracing work in Vercel edge functions?

### DISCUSSION THREAD

**Discussion by holdenmatt** [langfuse/langfuse#4389] [Support] [answered]
Has anyone gotten Langfuse working in Vercel "edge" functions?

I have a nextjs app hosted on Vercel using the "ai" sdk. I followed the guide here to setup Langfuse tracing using OpenTelemetry: https://langfuse.com/docs/integrations/vercel-ai-sdk

I see traces from my dev environment but none from my production app. After debugging a while, I disabled the "edge" runtime in my api route and production traces finally appeared.

If I use "edge", I don't see any issues in Vercel build or runtime logs, traces just never appear in Langfuse.

Is this expected?

**Comment by hassiebp** [comment]
Hi @holdenmatt! Could you please enable debug logs and share whether you see any logs / spans exported in your edge runtime? Also, could you please add a call to the `forceFlush` method on the LangfuseExporter instance before returning in your handler?

```ts
new LangfuseExporter({ debug: true })
```

**Comment by RobertBroersma** [comment]
I'm running into the same issue.

I've enabled debug logs in my LangfuseExporter and I can see `span-create`, `generation-create` and `trace-create` in my logs on Vercel.

I'm using `streamText` from the Vercel AI SDK and I've added `await langfuseExporter.forceFlush()` to the end of the `onFinish` callback. I can see the
> [LangfuseExporter] Force flushing Langfuse...

in my logs yet nothing appears in my dashboard.

I've also tried using the `waitUntil` feature from `@vercel/functions` in the `onFinish` callback.

Exact same setup works locally.

EDIT: For me it's not working in the node.js runtime either on Vercel :(

**Reply by hassiebp** [reply]
@RobertBroersma In debug mode, do you see exported spans in your console? Are spans traced correctly in dev?

**Reply by RobertBroersma** [reply]
@hassiebp not 100% sure what you mean. I do see lines that mention exported spans.

Everything works in dev - locally. I see the traces appearing in Langfuse.

**Reply by hassiebp** [reply, marked as answer]
@RobertBroersma Can you please verify whether the environment variables for Langfuse host / secret key and public key are set correctly in your Vercel project and available in your edge runtime by logging them out (and rotating them later)? 

The integration to not work on edge *and* node runtimes suggest that this might be rather an issue in your setup. If indeed all is setup correctly, could you also share a redacted version of your debug logs with me either here or to hassieb at langfuse .com ?
